### PR TITLE
Refresh the Jira Access Request plugin guide

### LINKED ISSUE #14580: Refresh the JIRA Cloud access requests guide
## Details
See #14236

### Category
- Improve Existing

### LINKED ISSUE #17528: Add `expired` state to jira workflow
## Applies To

Add `expired` state to workflow:

https://goteleport.com/docs/access-controls/access-request-plugins/ssh-approval-jira-server/#setting-up-the-status-board

## Details

Jira plugin supports transition to `expired` state from `pending` state, if a request is not approved or if its `canceled` by requester.

The document above should have `expired` state too.

## Related Issues

- #14580

### PR DESCRIPTION
**Pull request by ptgott**
Closes #14580
Closes #17528

- Get the Jira guide to follow the structure of other self-hosted Access Request plugin guides (cloud-hosted Access Request plugin instructions will arrive in a separate PR).
- Add instructions for obtaining TLS credentials for the plugin, both for Linux VMs and Kubernetes clusters.
- Add a diagram.

### CODE REVIEW

File: examples/resources/plugins/teleport-jira-cloud.toml
Comment (lsgunn-teleport) [inline examples/resources/plugins/teleport-jira-cloud.toml:15]:
```suggestion
# URL on which webhook server is accessible externally, for example,
```

**Review by lsgunn-teleport — APPROVED** [review]

File: docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx
Comment (tcsc) [inline docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx:13]: FYI - the webhook server is disabled in cloud-hosted plugins, due to the added complexity of trying to get the callback requests back through the Teleport proxy to an arbitrary server inside the cluster. 

It's more of an issue for my #30254 PR, but it would be a big help if things are structured so that any mentions of the webhook can be easily omitted in the `cloud` scope

**Comment by ptgott** [comment]
I'm going to hold off on getting this PR merged until we merge the docs for the managed Jira plugin: https://github.com/gravitational/teleport/pull/30254

File: docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx
Comment (ptgott) [reply docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx:13]: The page now includes a partial that refers Cloud users to the Web UI, which hopefully removes some of the confusion.

File: docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx
Comment (zmb3) [inline docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx:131]: Are we actually using Caddy as a web server or are we just using it to get certs? If we're not using it as a web server, than woudln't something like certbot make more sense?

Comment (zmb3) [reply docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx:125]: I'm still a little confused. We didn't mention Caddy before, so what has changed that now requires it?

It's not necessarily a bad thing, the barrier to entry gets higher which each extra piece of OSS you have to cobble together to get the system up and running.

Comment (ptgott) [reply docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx:125]: Caddy itself isn't required, but the user needs some way to provide TLS credentials to the Jira webhook. This requirement hasn't changed since we published the original guide, which [mentions it briefly](https://github.com/gravitational/teleport/blob/5df13390fd57eaf5338dc5cea92f8272fc27cc30/docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx#L136-L137) in the configuration section. With this PR, I wanted to help the user remove some decision making in order to get to the objective of the guide more quickly, so I added instructions for getting TLS credentials with `cert-manager` and Caddy. 

I attempted to [provide some context](https://github.com/gravitational/teleport/blob/abcbdcb22ab7812159d3d255061823dc0d1bf020/docs/pages/access-controls/access-request-plugins/ssh-approval-jira.mdx#L90-L91) around the TLS examples, but maybe I could be more explicit about why we're including Caddy/cert-manager instructions?

**Comment by zmb3** [comment]
@ptgott do we still need this now that #35635 has merged?

**Comment by ptgott** [comment]
@zmb3 I'd like to do something with the cert-manager and Caddy instructions before I delete my branch and close this PR. Either I'll:
- open a new PR to include these instructions with proper signposting so it's clear to users that these specific technologies aren't required, but rather mentioned to illustrate a quick, copy-pasteable approach for demo deployments.
- include these instructions in internal documentation to make it easier for me/others to test this guide in the future.

**Pull request closed without merging**
